src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/util/WaitFor.java

```diff
@@ -16,16 +18,31 @@
      * @throws Exception If the time limit expires before the task evaluates to true
      */
     public static void waitFor(Callable<Boolean> task, long maxWait, TimeUnit unit) throws Exception {
+        waitFor(task, maxWait, unit, "waitFor timed out before task evaluated to true");
+    }
+
+    /**
+     * Waits at most the specified time for the given task to evaluate to true
+     * @param task The task which we hope evaluates to true before the time limit
+     * @param maxWait The max amount of time to wait for the task to evaluate for true
+     * @param unit The time unit of the maxWait parameter
+     * @param message The exception message to print if waitFor times out
+     * @throws Exception If the time limit expires before the task evaluates to true
+     */
+    public static void waitFor(Callable<Boolean> task, long maxWait, TimeUnit unit, String message)
```

Comment:
Can we do it in the same function? Do we need method overloading for a single hardcoded error string?

Reply:
You're right, right now this isn't being used anywhere. However, I it's valuable for developers to be able to print their own messages context if their routine fails.